### DISCUSSION TITLE
Why are Param Functions uppercased?

### DISCUSSION THREAD

**Discussion by aryaniyaps** [fastapi/fastapi#6473] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends

app = FastAPI()

def get_dep():
    pass


@app.get("/)
def my_route(dep = Depends(get_dep)):
    pass
```

### Description

I really like FastAPI's dependency injection system. It makes it really easy to write scalable apps with less code. However, I dont understand why param functions like Depends and Security are uppercased.

After all, they are functions and this violates PEP8.

I find it more pythonic to make the function names lowercase, although I'm sure there would be a reason behind this decision.

Example:
```py
from fastapi import FastAPI, depends

app = FastAPI()

def get_dep():
    pass


@app.get("/)
def my_route(dep = depends(get_dep)):
    pass
```

I would like to hear the thought process behind this from the maintainers!

### Operating System

Windows

### FastAPI Version

0.70

### Python Version

3.9

**Comment by dabula-s** [comment]
> However, I dont understand why param functions like Depends and Security are uppercased.

actually `Depends` is a class... not a function. and also `Security` is.

**Comment by aryaniyaps** [comment]
@SeriyVol4ishe I had actually meant these functions, which return those classes:

https://github.com/tiangolo/fastapi/blob/master/fastapi/param_functions.py

**Comment by Kludex** [comment]
Afaik, to satisfy type checkers.

**Comment by insomnes** [comment, marked as answer]
From docs [here](https://fastapi.tiangolo.com/tutorial/path-params-numeric-validations/#recap)

> When you import Query, Path and others from fastapi, they are actually functions.
> That when called, return instances of classes of the same name.
> So, you import Query, which is a function. And when you call it, it returns an instance of a class also named Query.
> These functions are there (instead of just using the classes directly) so that your editor doesn't mark errors about their types.
> That way you can use your normal editor and coding tools without having to add custom configurations to disregard those errors.

**Comment by aryaniyaps** [comment]
@insomnes thanks a lot! I think this solves the issue!

**Comment by insomnes** [comment]
@aryaniyaps great to hear! Please consider to close the issue if your question is answered
